packages/babel-helper-define-map/src/index.js

```diff
@@ -62,7 +62,7 @@ export function push(mutatorMap: Object, node: Object, kind: string, file, scope
 
   // infer function name
   if (scope && t.isStringLiteral(key) && (kind === "value" || kind === "initializer") && t.isFunctionExpression(value)) {
-    value = nameFunction({ id: key, node: value, scope });
+    value = nameFunction({ id: key, node: value, scope, isClassMethod: t.isClassMethod(node) });
```

Comment:
Yeah maybe we could potentially pass the node or the type instead of checking if it is a `ClassMethod` specifically